src/utils/IO/system_checks.jl

```diff
@@ -1,34 +1,27 @@
 
 ### Utility Functions needed for the construction of the Power System, mostly used for consistency checking ####
 
-## Time Series Length ##
-
-function timeseriescheckforecast(forecasts::Forecasts)
-    t = length(unique([length(f) for f in forecasts]))
-    if t > 1
-        @error "forecast array contains $t different time series lengths"
-    end
-end
-
 ## Check that all the buses have a type defintion ##
 
-function buscheck(buses::Array{Bus})
+function buscheck(buses)
     for b in buses
-        if b.bustype == nothing
+        if isnothing(b.bustype)
             @warn "Bus/Nodes data does not contain information to build an a network"
         end
     end
 
-    check_ascending_order([x.number for x in buses], "Bus")
+    # TODO DT: do we need this?
```

Comment:
Based on the change to storing components in dictionaries instead of arrays we can no longer guarantee ordering.  If this is OK, I will delete this check.  If we do want ordering then we could do one of the following:
1. Store the components in an OrderedDict or SortedDict.
2. Change get_components so that it returns a sorted array.  This would be much less efficient than what we have now.
3. Create a new function that performs the second option.

Let me know what you think.

Reply:
Not ordering is ok. No models should rely on order to work. If something broke, as a result, the fix is in the model itself.

Reply:
Removed.